src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -920,14 +920,18 @@ protected internal void OnDocumentInfoChanged(DocumentId documentId, DocumentInf
         /// Call this method when the text of a document is updated in the host environment.
         /// </summary>
         protected internal void OnDocumentTextChanged(DocumentId documentId, SourceText newText, PreservationMode mode)
+            => OnDocumentTextChanged(documentId, newText, mode, requireDocumentPresent: true);
+
+        private protected void OnDocumentTextChanged(DocumentId documentId, SourceText newText, PreservationMode mode, bool requireDocumentPresent)
```

Comment:
probably would be good to add docs for these new methods indicating that you might use them when you don't want to throw if it fails